tests/unit/Hooks/ImageHooksTest.php

```diff
@@ -0,0 +1,224 @@
+<?php
+/**
+ * ownCloud - mediametadata
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author Jalpreet Singh Nanda (:imjalpreet) <jalpreetnanda@gmail.com>
+ * @copyright Jalpreet Singh Nanda (:imjalpreet) 2016
+ */
+
+namespace OCA\MediaMetadata\Hooks;
+
+function getimagesize($filename = null, array &$imageinfo = null) {
+	return ImageHooksTest::$dimensions ?: getimagesize($filename);
```

Comment:
Ah.. This was the reason for the segfault. But now I am getting this expection failed error: [link](https://gist.github.com/imjalpreet/54e6ce7ddff35455a370ab9238149830)
Is this because the way `entity` is mocked is wrong?